slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidator.java

```diff
@@ -43,10 +46,23 @@ public GenesisValidatorRootValidator(final Jdbi jdbi, final MetadataDao metadata
   }
 
   public boolean checkGenesisValidatorsRootAndInsertIfEmpty(final Bytes32 genesisValidatorsRoot) {
-    return failsafeExecutor.get(
-        () ->
-            jdbi.inTransaction(
-                READ_COMMITTED, handle -> validateGvr(handle, genesisValidatorsRoot)));
+    if (cachedGenesisValidatorRoot == null) {
+      cachedGenesisValidatorRoot =
+          failsafeExecutor.get(
+              () ->
+                  jdbi.inTransaction(
+                      READ_COMMITTED,
+                      handle -> findAndInsertIfNotExists(handle, genesisValidatorsRoot)));
+    }
+
+    if (Objects.equals(cachedGenesisValidatorRoot, genesisValidatorsRoot)) {
```

Comment:
Why we do need to use Object.equals here?

Reply:
its to avoid NPE in case cachedGenesisValidatorRoot is still null after the insert call.